### DISCUSSION TITLE
PicoCNC - Machine is stuck in negative coordinate space and can't seem to force set origin on homing

### DISCUSSION THREAD

**Discussion by mrrossdude** [grblHAL/RP2040#97] [Q&A] [answered]
Hi, 

I've upgraded a Makerdreams Evo ONE hobby cnc with the PicoCNC board. The machine was upgraded with spindle/nema 23 and external stepper drivers, and has been running well for two years. I've upgraded to the PicoCNC as the original board failed, and I wanted relay controls/ethernet etc. 

All axes jog correctly and as expected when looking at the "d-pad" jog controller in IOsender/UGS - where X+ moves spindle to the right, Y+ moves the table towards you (so spindle is at the back, and Z+ moves spindle up. 

The machine is a fixed gantry, and it homes to limit switches so that the spindle is at the top/back left of the spoilboard, which it does correctly. Originally, the machine used to zero itself after homing, and then you'd move to the workpiece and zero to it for G54 coordinates. At the end of a CAM process, it would return to G53 0,0,0. All my cam files are processed from fusion in this way and it makes sense to my brain so I'd like to keep it.

Problem is that the default option in the PicoCNC firmware seems to have it set to negative space, so now after homing the coordinates are X-367.3, Y-0.5, Z-0.5. As a result, none of my macros work and I can't use my existing CAM processes. I also don't seem to be able to actually create zeroes for G54. Pressing the zero axis buttons in IOsender/UGS does nothing. 

There is no option I can see in the WebBuilder for this, so I've cloned the RP2040 repository and done the following changes:

in my_machine.h, uncomment:
	#define BOARD_PICO_CNC
	#define ETHERNET_ENABLE   0  (but don't enable here according to comment)
	#define _WIZCHIP_   5500
	
in CmakeLists.txt:
	option(ADD_ETHERNET "Add Ethernet networking" ON)

in Config.h:
	DEFAULT_HOMING_FORCE_SET_ORIGIN changed to ON

I'm making these changes in PicoVSCode, and then building from the CMakeLists.txt in the root folder, and uploading the firmware to the Pico with Bootsel/reset method over USB. 

However, my machine still shows to be in negative space after homing. I am completely new to any form of programming so I could be doing it completely wrong. 

Any ideas? Alternatively, could anyone provide a built firmware for the PicoCNC, but with Ethernet enabled and forced zero after homing?

Thanks in advance

**Comment by terjeio** [comment, marked as answer]
Default is to home in negative space, but you can set it to home to 0 in the settings:

![image](https://github.com/user-attachments/assets/092e31db-511e-40d9-aecc-066fef8b3db0)

**Reply by terjeio** [reply]
> DEFAULT_HOMING_FORCE_SET_ORIGIN changed to ON

FYI if you change defaults in _config.h_ you have to [reset the settings](https://github.com/grblHAL/core/blob/c07d63198e2af5e9895e2848bb33e1eb7cc35aff/config.h#L26-L29) after flashing.
